transaction/transactor_test.go

```diff
@@ -134,3 +139,70 @@ func TestWelcome(t *testing.T) {
 	req := httptest.NewRequest(http.MethodGet, "http://localhost", nil)
 	decorated.ServeHTTP(nil, req)
 }
+
+func TestLog(t *testing.T) {
+	ctxWithArrivalTime := context.WithValue(context.Background(), ContextKeyRequestArrivalTime, time.Date(2009, time.November, 10, 23, 0, 0, 0, time.UTC))
+	logCount := 0
+	tcs := []struct {
+		desc                        string
+		logger                      *zap.Logger
+		reducedLoggingResponseCodes []int
+		ctx                         context.Context
+		code                        int
+		request                     *http.Request
+		expectedLogCount            int
+	}{
+		{
+			desc: "Sanity Check",
+			logger: zaptest.NewLogger(t, zaptest.WrapOptions(zap.Hooks(
+				func(e zapcore.Entry) error {
+					if e.Level == zap.ErrorLevel {
+						t.Fatal("Error should never happen!")
+					}
+					logCount++
+					return nil
+				}))),
+			reducedLoggingResponseCodes: []int{},
+			ctx:                         context.Background(),
+			code:                        200,
+			request:                     &http.Request{},
+			expectedLogCount:            1,
+		},
+		{
+			desc: "Arrival Time Present",
+			logger: zaptest.NewLogger(t, zaptest.WrapOptions(zap.Hooks(
+				func(e zapcore.Entry) error {
+					logCount++
+					return nil
+				}))),
+			reducedLoggingResponseCodes: []int{},
+			ctx:                         ctxWithArrivalTime,
+			code:                        200,
+			request:                     &http.Request{},
+			expectedLogCount:            2,
+		},
+		{
+			desc: "IncludeHeaders is False",
+			logger: zaptest.NewLogger(t, zaptest.WrapOptions(zap.Hooks(
+				func(e zapcore.Entry) error {
+					logCount++
+					return nil
+				}))),
+			reducedLoggingResponseCodes: []int{200},
+			ctx:                         context.Background(),
+			code:                        200,
+			request:                     &http.Request{},
+			expectedLogCount:            1,
+		},
+	}
+
+	for _, tc := range tcs {
+		t.Run(tc.desc, func(t *testing.T) {
+			assert := assert.New(t)
+			logCount = 0
```

Comment:
Ideally, we don't want our separate test cases to have shared memory.  If these tests were changed in the future to run in parallel, it wouldn't work.  Since this is easy to fix, I think we should.  We can either:
1. Define logCount exclusively within the `t.Run()` function and then define the anonymous function here as well.  Then, since it's a new log count for every test, we know they're not affecting one another. 
2. Create a struct that holds the count in it.  Then the tests can expect an interface and be more flexible in the future.

I've thought about the details of 2 and I'm not sure that the implementation cost is worth the potential flexibility needed in the future.  Personally, I would rather do option 1 and then if we need something more in the future we can build it then.

Reply:
Option 1 would look like:
```suggestion
			logCount := 0
			logger := zaptest.NewLogger(t, zaptest.WrapOptions(zap.Hooks(
				func(e zapcore.Entry) error {
					logCount++
					return nil
				})))
```